test-app/tests/integration/components/multiselect-test.gts

```diff
@@ -0,0 +1,1215 @@
+
+    // Filtering works as we expect
+    assert.dom('[role="option"]').exists({ count: 1 });
+    // NOTE: We should be using option.label (capitalized "Blue" instead of "blue")
+    assert.dom('[role="option"]').hasText('Blue');
+
+    // Resetting the filter by clearing the input should
+    // display all available options
+    await fillIn('[data-multiselect]', '');
+    assert.dom('[role="option"]').exists({ count: 2 });
+
+    // Verify we can filter again after clearing
+    await fillIn('[data-multiselect]', 'red');
+    assert.dom('[role="option"]').exists({ count: 1 });
+    // NOTE: We should be using option.label (capitalized "Red" instead of "red")
+    assert.dom('[role="option"]').hasText('Red');
+  });
+
+  test('it uses the provided `@noResultsText` when no results are found with filtering', async function (assert) {
+    await render(<template>
+      <Multiselect
+        @options={{testColors}}
+        @noResultsText="No results"
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    await fillIn('[data-multiselect]', 'something-not-in-the-list');
+
+    // We should not have any list items
+    assert.dom('[role="option"]').exists({ count: 0 });
+
+    // ...but we should have our no results item!
+    assert.dom('[role="status"]').exists();
+    assert.dom('[role="status"]').hasTagName('li');
+    assert.dom('[role="status"]').hasText('No results');
+    assert
+      .dom('[role="status"]')
+      .hasAttribute(
+        'aria-live',
+        'assertive',
+        'Expected assertive so it is announced to screenreaders'
+      );
+  });
+
+  test('it calls `@onChange` when an option is selected via mouse click and keeps the popover oepn', async function (assert) {
+    assert.expect(6);
+
+    let handleChange = (value: string[]) => {
+      assert.deepEqual(value, ['blue'], 'Expected input to match');
+      assert.step('handleChange');
+    };
+
+    await render(<template>
+      <Multiselect
+        @options={{testColors}}
+        @onChange={{handleChange}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option data-option>
+          {{multiselect.option}}
+        </multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    assert.verifySteps([]);
+
+    await fillIn('[data-multiselect]', 'blue');
+
+    assert.dom('[role="option"]').exists({ count: 1 });
+
+    await click('[role="option"]');
+
+    assert.verifySteps(['handleChange']);
+
+    // Verify the popover remains open
+    assert.dom('[role="listbox"]').exists();
+  });
+
+  test('it calls `@onChange` when an option is selected via the keyboard with ENTER and keeps the popover open', async function (assert) {
```

Comment:
This tests for a single option, can the text be expanded to ensure multiple options can be selected @onChange...or does it call @onChange per option?

Reply:
It calls `@onChange` per option!  And I have a few tests sprinkled throughout that start with `@selected` options and then add / remove, so `@onChange` gets exercised pretty thoroughly